micro-rdk/src/common/conn/server.rs

```diff
@@ -296,40 +325,81 @@ where
         app_connector: C,
         app_config: AppClientConfig,
         max_concurent_connections: usize,
+        app_client_tasks: Vec<Box<dyn PeriodicAppClientTask>>,
     ) -> Self {
         Self {
             http_listener,
             webrtc_config,
             exec,
             app_connector,
             app_config,
-            app_client: None,
+            app_client: Default::default(),
             incoming_connection_manager: IncomingConnectionManager::new(max_concurent_connections),
+            app_client_tasks,
         }
     }
     pub async fn serve(&mut self, robot: Arc<Mutex<LocalRobot>>) {
         let cloned_robot = robot.clone();
+
+        let _app_client_tasks: Vec<_> = self
+            .app_client_tasks
+            .drain(..)
+            .map(|mut task| {
+                let app_client = Rc::clone(&self.app_client);
+                self.exec.spawn(async move {
+                    let mut duration = task.get_default_period();
+                    loop {
+                        let _ = async_io::Timer::after(duration).await;
+                        let mut app_client_guard = app_client.lock().await;
+                        if app_client_guard.is_none() {
+                        } else {
+                            match task.invoke(app_client_guard.as_mut().unwrap()).await {
+                                Ok(None) => continue,
+                                Ok(Some(next_duration)) => {
```

Comment:
This machinery around duration tracking was mostly to serve what appears to be the intention of the restart endpoint to be able to reply with a different timeout for the next check. It certainly adds some complexity so I'd be OK finding another way.